### PR TITLE
fix(Scripts/DurnholdeKeep): Captain Skarloc spawning double/wrong adds

### PR DESCRIPTION
**Pull request by avarishd**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/6008
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17141

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Kalimdor/CavernsOfTime/EscapeFromDurnholdeKeep/boss_captain_skarloc.cpp [hunk 1]
```diff
@@ -108,27 +110,31 @@ struct boss_captain_skarloc : public BossAI
         if (type != ESCORT_MOTION_TYPE)
             return;
 
-        // Xinef: we can rely here on internal counting
-        if (id == 1)
+        if (!_SpawnedAdds)
         {
```
Comment (elthehablo) [inline src/server/scripts/Kalimdor/CavernsOfTime/EscapeFromDurnholdeKeep/boss_captain_skarloc.cpp:114]: very nerdy comment, but variables in C++ are camelCase and methods are PascalCase

File: src/server/scripts/Kalimdor/CavernsOfTime/EscapeFromDurnholdeKeep/boss_captain_skarloc.cpp [hunk 2]
```diff
@@ -108,27 +110,31 @@ struct boss_captain_skarloc : public BossAI
...
+                me->SummonCreature(NPC_DURNHOLDE_VETERAN, 2032.810f, 269.416f, 63.561f, 5.30f, TEMPSUMMON_MANUAL_DESPAWN);
             }
-            if (Creature* mount = me->SummonCreature(NPC_SKARLOC_MOUNT, 2049.12f, 252.31f, 62.855f, me->GetOrientation(), TEMPSUMMON_MANUAL_DESPAWN))
+            else if (id == 2)
             {
-                mount->SetImmuneToNPC(true);
-                mount->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+                me->Dismount();
+                me->SetWalk(true);
+                for (SummonList::const_iterator itr = summons.begin(); itr != summons.end(); ++itr)
```
Comment (elthehablo) [inline src/server/scripts/Kalimdor/CavernsOfTime/EscapeFromDurnholdeKeep/boss_captain_skarloc.cpp:124]: summons.DoForAllSummons()

**Review by elthehablo — CHANGES_REQUESTED** [review]

File: src/server/scripts/Kalimdor/CavernsOfTime/EscapeFromDurnholdeKeep/boss_captain_skarloc.cpp (hunk 2 above)
Comment (avarishd) [reply src/server/scripts/Kalimdor/CavernsOfTime/EscapeFromDurnholdeKeep/boss_captain_skarloc.cpp:124]: https://media.tenor.com/fHz7wfvakO8AAAAC/theyre-the-same-picture-the-office.gif

Comment (elthehablo) [reply src/server/scripts/Kalimdor/CavernsOfTime/EscapeFromDurnholdeKeep/boss_captain_skarloc.cpp:124]: cleans up the code. but ye does the same essentially. please use it tho

**Review by elthehablo — APPROVED** [review]

**Pull request merged**